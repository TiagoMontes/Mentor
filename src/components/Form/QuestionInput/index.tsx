import Field from '@/components/Field'
import { UseFormRegister } from 'react-hook-form'
import { FormData } from '@/type'

type QuestionInputProps = {
  register: UseFormRegister<FormData>
}

export default function QuestionInput({ register }: QuestionInputProps) {
  return (
    <>
      <Field
        inputName="questionAsk"
        register={register}
        placeholder="FaÃ§a uma pergunta..."
        className="bg-black outline-none"
      />
    </>
  )
}
