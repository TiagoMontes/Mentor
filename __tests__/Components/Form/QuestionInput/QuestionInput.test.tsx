import '@testing-library/jest-dom'
import { render, screen } from '@testing-library/react'
import QuestionInput from '@/components/Form/QuestionInput'

const mockRegister = jest.fn().mockReturnValue({
  name: 'questionAsk',
  onChange: jest.fn(),
  onBlur: jest.fn(),
  ref: jest.fn(),
})

describe('QuestionInput', () => {
  beforeEach(() => {
    jest.clearAllMocks()
  })

  it('renders the QuestionInput component with an input', () => {
    render(<QuestionInput register={mockRegister} />)
    const input = screen.getByPlaceholderText('Faça uma pergunta...')
    expect(input).toBeInTheDocument()
  })

  it('applies the correct props to the input', () => {
    render(<QuestionInput register={mockRegister} />)
    const input = screen.getByPlaceholderText('Faça uma pergunta...')

    expect(input).toHaveClass(
      'bg-black outline-none w-full rounded-lg border-2 border-black p-2'
    )

    expect(input).toHaveAttribute('type', 'text')

    expect(input).toHaveAttribute('placeholder', 'Faça uma pergunta...')

    expect(mockRegister).toHaveBeenCalledWith('questionAsk', {})
  })

  it('should check if the total number of inputs rendered is as expected', () => {
    render(<QuestionInput register={mockRegister} />)
    const allInputs = screen.queryAllByRole('input')

    expect(allInputs).toBe(1)
  })

  it('matches snapshot', () => {
    const { container } = render(<QuestionInput register={mockRegister} />)
    expect(container).toMatchSnapshot()
  })
})
