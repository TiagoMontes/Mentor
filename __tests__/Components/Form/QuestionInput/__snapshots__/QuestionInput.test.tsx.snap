// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuestionInput matches snapshot 1`] = `
<div>
  <input
    class="bg-black outline-none w-full rounded-lg border-2 border-black p-2"
    name="questionAsk"
    placeholder="FaÃ§a uma pergunta..."
    type="text"
  />
</div>
`;
